src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py

```diff
@@ -0,0 +1,796 @@
+
+        Returns:
+            The registered model version.
+        """
+        # Check if the model exists, if not create it.
+        if not self.check_model_exists(name):
+            logger.warning(
+                f"No registered model with name {name} found. Creating a new"
+                "registered model."
+            )
+            self.register_model(
+                name=name,
+                description=description,
+                tags=tags,
+            )
+        try:
+            # Inform the user that the version is ignored.
+            if version:
+                logger.info(
+                    f"MLflow model registry does not take a version as an argument. "
+                    f"Registering a new version for the model `'{name}'` "
+                    f"a version will be assigned automatically."
+                )
+            # Set the tags.
+            if not version_tags:
+                version_tags = {}
+            version_tags["zenml_version"] = zenml_version or __version__
+            version_tags["zenml_pipeline_run_id"] = zenml_pipeline_run_id or ""
+            version_tags["zenml_pipeline_name"] = zenml_pipeline_name or ""
+            version_tags["zenml_step_name"] = zenml_step_name or ""
+            # Register the model version.
+            registered_model_version = self.mlflow_client.create_model_version(
+                name=name,
+                source=model_source_uri,
+                run_id=registry_metadata.get("mlflow_run_id")
+                if registry_metadata
+                else "",
+                run_link=registry_metadata.get("mlflow_run_link")
+                if registry_metadata
+                else "",
+                description=version_description,
+                tags=version_tags,
+            )
+        except MlflowException as e:
+            logger.error(
+                f"Failed to register new model version for model '{name}'."
+            )
+            raise e
+        # Return the registered model version.
+        return ModelVersion(
+            model_registration=ModelRegistration(
+                name=registered_model_version.name,
+                description=description,
+                tags=tags,
+            ),
+            model_source_uri=registered_model_version.source,
+            registry_metadata={
+                "mlflow_run_id": registered_model_version.run_id or "",
+                "mlflow_run_link": registered_model_version.run_link or "",
+            },
+            version=registered_model_version.version,
+            description=registered_model_version.description,
+            version_stage=registered_model_version.current_stage,
+            tags=registered_model_version.tags,
+            created_at=str(registered_model_version.creation_timestamp),
+            last_updated_at=str(
+                registered_model_version.last_updated_timestamp
+            ),
```

Comment:
wouldn't it be better to keep these as `datetime.datetime` ? Then, maybe you could also figure out what the timezone was.